### PR TITLE
PLT-4935

### PR DESCRIPTION
**Pull request by mprasoonquartic**
Modify to handle pagination based upon cursor instead of offset

### CODE REVIEW

File: quartic_sdk/core/iterators/tag_data_iterator.py [hunk 1]
```diff
@@ -77,22 +80,26 @@ def raise_exception_for_transformation_schema(transformations, tags):
...
         if len(agg_transformation) > 1:
-            raise Exception("Invalid transformations : Only one aggregation transformation can be applied at a time")
+            raise Exception(
+                "Invalid transformations : Only one aggregation transformation can be applied at a time")
         for transformation in transformations:
             transformation_type = transformation.get("transformation_type")
             if transformation_type == "interpolation":
                 if transformation.get("column") is None:
-                    raise Exception("Invalid transformations : Interpolation column is missing")
+                    raise Exception(
```
Comment (pr4n) [inline quartic_sdk/core/iterators/tag_data_iterator.py:89]: `Exception` is a very high level class. Typically one does subclass it with a more reasonable name like `class InvalidTransformationException(Exception): pass` and then use this Exception. This makes it's catch better and more relevant.

**Review by pr4n — CHANGES_REQUESTED** [review]

File: quartic_sdk/core/iterators/tag_data_iterator.py (hunk 1 above)
Comment (mprasoonquartic) [reply quartic_sdk/core/iterators/tag_data_iterator.py]: done

**Review by KishanKishore — COMMENTED** [review]
Could add more details to the documentation strings.

File: quartic_sdk/core/entities/asset.py [hunk 2]
```diff
@@ -28,8 +28,15 @@ def get_tags(self):
...
+            Constants.API_GET,
+            path_params=[],
+            query_params={
+                "asset": self.id}).json()
+        return EntityFactory(
+            Constants.TAG_ENTITY,
+            tags_response,
+            self.api_helper)
 
     def event_frames(self):
```
[the remark this answers is not in the document]
Comment (mprasoonquartic) [reply quartic_sdk/core/entities/asset.py:41]: Providing that will leead to circular ddependency error. heence removed all

File: quartic_sdk/core/entities/edge_connector.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ def get_tags(self):
             Constants.GET_TAGS, Constants.API_GET, path_params=[], query_params={"edge_connector": self.id}).json()
         return EntityFactory(Constants.TAG_ENTITY, tags_response, self.api_helper)
 
-    def data(self, start_time, stop_time, granularity=0, return_type=Constants.RETURN_PANDAS, transformations=[]):
+    def data(self, start_time, stop_time, granularity=0, return_type=Constants.RETURN_PANDAS, batch_size=Constants.DEFAULT_PAGE_LIMIT_ROWS, transformations=[]):
```
[the remark this answers is not in the document]
Comment (mprasoonquartic) [reply quartic_sdk/core/entities/edge_connector.py:33]: Providing that will leead to circular ddependency error. heence removed all

**Review by scriperdjq — DISMISSED** [review]

File: quartic_sdk/core/entities/asset.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (KishanKishore) [reply quartic_sdk/core/entities/asset.py:41]: Mention it in the documentation string.

File: quartic_sdk/core/iterators/tag_data_iterator.py [hunk 4]
```diff
@@ -103,58 +111,49 @@ def create_post_data(self):
...
-
-        return tag_data_return
+        self._data_call_state = 0
+        return self
 
-    def __getitem__(self, key):
+    def __next__(self):
         """
-        We override this method to get the object at the given key
+        Get the next object in the iteration.
```
Comment (KishanKishore) [inline quartic_sdk/core/iterators/tag_data_iterator.py:127]: return type is missing documentation string.

File: quartic_sdk/core/iterators/tag_data_iterator.py [hunk 5]
```diff
@@ -190,24 +190,44 @@ def create_tag_data_iterator(
...
-            count=tag_data_response["count"],
+            total_count=tag_data_response["total_count"],
             api_helper=api_helper,
+            batch_size=batch_size,
             granularity=granularity,
             return_type=return_type,
             transformations=transformations)
+
+    @staticmethod
+    def return_complete_data_frame_from_iterator(tag_data_iterator):
```
Comment (KishanKishore) [inline quartic_sdk/core/iterators/tag_data_iterator.py:220]: Provide the return type here.

**Review by KishanKishore — COMMENTED** [review]
Minor comments related to documentation strings and return types.

File: quartic_sdk/core/entities/asset.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mprasoonquartic) [reply quartic_sdk/core/entities/asset.py:41]: will be adding the type hints separately when updating the docs

File: quartic_sdk/core/iterators/tag_data_iterator.py (hunk 5 above)
Comment (mprasoonquartic) [reply quartic_sdk/core/iterators/tag_data_iterator.py:220]: let me handle this seperately while updating the docs

**Review by pr4n — APPROVED** [review]

**Review by scriperdjq — APPROVED** [review]

**Pull request merged**